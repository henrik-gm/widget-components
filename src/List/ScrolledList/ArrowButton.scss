:local(.left),
:local(.right) {
   position: absolute;
   top: 0;
   bottom: 0;
   width: 35px;
   border: none;
   background: none;
   font-size: 15px;
   font-weight: bold;
   cursor: pointer;
   outline: none;
   padding: 0;
   display: flex;
   align-items: center;

   &[disabled] {
      display: none;
   }

   :local(.icon) {
      fill: #000;
      opacity: .3;
      transition: opacity .3s;
   }

   &:hover :local(.icon) {
      opacity: 1;
   }
}

:local(.left) {
   left: 16px;
}

:local(.right) {
   right: 16px;

   :local(.icon) {
      transform: scale(-1, 1);
      margin-left: 25px;
   }
}
