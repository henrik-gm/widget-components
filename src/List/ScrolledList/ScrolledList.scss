:local(.container) {
   position: relative;
   padding: 0 16px;

   &.mobile-device {
      padding: 0;
   }
}

// field of view
:local(.eyeshot) {
   overflow-x: hidden;

   .mobile-device & {
      overflow-x: scroll;
      -webkit-overflow-scrolling: touch;
   }
}

// tab bar
:local(.bar) {
   display: flex;
   margin: 0;
   transition: transform .3s;

   &.no-transition {
      transition: transform 0s;
   }

   // let tab containers manage their width
   & > * {
      flex: 0 0 auto;
   }
}
