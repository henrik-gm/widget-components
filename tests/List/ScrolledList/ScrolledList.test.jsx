/* eslint-env jest */

import React from 'react';
import ScrolledList from '../../../src/List/ScrolledList/ScrolledList';
import ReactTestRenderer from 'react-test-renderer'; // eslint-disable-line
import { shallow, mount } from 'enzyme'; // eslint-disable-line


describe('ScrolledList DOM rendering', () => {

   it('renders correctly with default props', () => {
      const tree = ReactTestRenderer.create(
         <ScrolledList />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with one child item', () => {
      const tree = ReactTestRenderer.create(
         <ScrolledList>
            <div />
         </ScrolledList>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with many child items', () => {
      const tree = ReactTestRenderer.create(
         <ScrolledList>
            <div>1</div>
            <div>2</div>
            {[<div key='3'>3</div>, <div key='4'>4</div>]}
         </ScrolledList>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with selected item given arbitrary', () => {
      const tree = ReactTestRenderer.create(
         <ScrolledList selected={1}>
            <div>1</div>
            <div>2</div>
         </ScrolledList>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly custom ItemContainer', () => {
      const tree = ReactTestRenderer.create(
         <ScrolledList
            renderItemContainer={({ selected, onClick, onWidth, children }) => <div>{children}</div>}
         >
            <div>1</div>
            <div>2</div>
         </ScrolledList>
      );

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly all item alignments', () => {
      Object.keys(ScrolledList.ALIGN_ITEMS).forEach((key) => {
         const tree = ReactTestRenderer.create(
            <ScrolledList alignItems={ScrolledList.ALIGN_ITEMS[key]}>
               <div>1</div>
               <div>2</div>
            </ScrolledList>
         ).toJSON();

         expect(tree).toMatchSnapshot();
      });
   });

   it('renders correctly in mobile mode with not known container width', () => {
      const oldUserAgent = window.navigator.userAgent;

      Object.defineProperty(window.navigator, 'userAgent', { get: () => 'iPhone', configurable: true });
      window.ontouchstart = () => {};

      const tree = ReactTestRenderer.create(
         <ScrolledList>
            <div>1</div>
            <div>2</div>
         </ScrolledList>
      );

      expect(tree).toMatchSnapshot();

      Object.defineProperty(window.navigator, 'userAgent', { get: () => oldUserAgent, configurable: true });
      delete window.ontouchstart;
   });

});

describe('ScrolledList behaviour', () => {

   it('calls onItemClick properly', () => {
      const onItemClickMock = jest.fn();

      const wrapper = mount(
         <ScrolledList onItemClick={onItemClickMock}>
            <div>1</div>
            <div>2</div>
            <div id='goingToBeClicked'>3</div>
         </ScrolledList>
      );

      expect(onItemClickMock).not.toHaveBeenCalled();

      wrapper.find('#goingToBeClicked').simulate('click');

      expect(onItemClickMock).toHaveBeenCalledTimes(1);
      expect(onItemClickMock).toHaveBeenCalledWith(2);
   });

   it('handles item clicks correctly', () => {
      const renderItemContainerMock = jest.fn(({ selected, onClick, onWidth, children }) => {
         return <div onClick={onClick}>{children}</div>;
      });

      const wrapper = mount(
         <ScrolledList renderItemContainer={renderItemContainerMock}>
            <div>1</div>
            <div>2</div>
            <div id='goingToBeClicked'>3</div>
         </ScrolledList>
      );

      expect(renderItemContainerMock).toHaveBeenCalledTimes(3);

      renderItemContainerMock.mockClear();

      wrapper.find('#goingToBeClicked').simulate('click');

      expect(renderItemContainerMock).toHaveBeenCalledTimes(3);
   });

   it('scrolls on arrows/item click correctly', () => {
      let nextButtonClick,
         prevButtonClick;

      const renderItemContainerMock = jest.fn(({ onClick, onWidth, children }) => {
         setTimeout(() => onWidth(100), 0);
         return <div onClick={onClick}>{children}</div>;
      });

      const renderNextButtonMock = jest.fn(({ onClick }) => {
         nextButtonClick = onClick;
         return <div />;
      });

      const renderPrevButtonMock = jest.fn(({ onClick }) => {
         prevButtonClick = onClick;
         return <div />;
      });

      const wrapper = mount(
         <ScrolledList
            renderItemContainer={renderItemContainerMock}
            renderNextButton={renderNextButtonMock}
            renderPrevButton={renderPrevButtonMock}
         >
            <div>1</div><div>2</div><div>3</div>
            <div>4</div><div>5</div><div>6</div>
            <div>7</div><div>8</div><div id='last'>9</div>
         </ScrolledList>
      );

      const eyeshotWrapper = wrapper.find('.eyeshot');
      eyeshotWrapper.node.offsetWidth = 300;

      const barWrapper = wrapper.find('.bar');

      return new Promise(resolve => setTimeout(() => resolve(), 0))
         .then(() => {
            if (!nextButtonClick) {
               throw new Error('nextButtonClick is undefined');
            }

            if (!prevButtonClick) {
               throw new Error('prevButtonClick is undefined');
            }

            expect(renderNextButtonMock).toHaveBeenCalled();
            expect(renderItemContainerMock).toHaveBeenCalled();
            expect(renderPrevButtonMock).toHaveBeenCalled();

            nextButtonClick();

            expect(eyeshotWrapper.node.scrollLeft).toEqual(0);
            expect(barWrapper.node.style.transform).toEqual('translate3d(-200px, 0, 0)');
            expect(barWrapper.node.style.mozTransform).toEqual('translate3d(-200px, 0, 0)');

            prevButtonClick();

            expect(eyeshotWrapper.node.scrollLeft).toEqual(0);
            expect(barWrapper.node.style.transform).toEqual('translate3d(0px, 0, 0)');
            expect(barWrapper.node.style.mozTransform).toEqual('translate3d(0px, 0, 0)');

            wrapper.find('#last').simulate('click');

            expect(eyeshotWrapper.node.scrollLeft).toEqual(0);
            expect(barWrapper.node.style.transform).toEqual('translate3d(0px, 0, 0)');
            expect(barWrapper.node.style.mozTransform).toEqual('translate3d(0px, 0, 0)');
         });
   });

   it('unmounts correctly', () => {
      const wrapper = mount(<ScrolledList />);
      wrapper.unmount();
   });

   it('aligns correctly when there are less items than container width', () => {
      const renderItemContainerMock = jest.fn(({ onClick, onWidth, children }) => {
         setTimeout(() => onWidth(100), 0);
         return <div onClick={onClick}>{children}</div>;
      });

      const wrapper = mount(
         <ScrolledList renderItemContainer={renderItemContainerMock}>
            <div>1</div><div>2</div>
         </ScrolledList>
      );

      const eyeshotWrapper = wrapper.find('.eyeshot');
      eyeshotWrapper.node.offsetWidth = 300;

      window.dispatchEvent(new Event('resize'));

      // test subsequent resize event
      window.dispatchEvent(new Event('resize'));

      const barWrapper = wrapper.find('.bar');

      return new Promise(resolve => setTimeout(() => resolve(), 210))
         .then(() => {
            expect(barWrapper.node.style.justifyContent).toEqual('center');

            eyeshotWrapper.node.offsetWidth = 100;

            window.dispatchEvent(new Event('resize'));

            return new Promise(resolve => setTimeout(() => resolve(), 210));
         })
         .then(() => expect(barWrapper.node.style.justifyContent).toEqual(''));
   });
});
