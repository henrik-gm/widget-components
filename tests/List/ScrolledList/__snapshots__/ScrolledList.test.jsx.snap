exports[`ScrolledList DOM rendering renders correctly all item alignments 1`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div
        className="item selected"
        onClick={[Function]}>
        <div>
          1
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item "
        onClick={[Function]}>
        <div>
          2
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
    </div>
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly all item alignments 2`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div
        className="item selected"
        onClick={[Function]}>
        <div>
          1
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item "
        onClick={[Function]}>
        <div>
          2
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
    </div>
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly all item alignments 3`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div
        className="item selected"
        onClick={[Function]}>
        <div>
          1
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item "
        onClick={[Function]}>
        <div>
          2
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
    </div>
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly all item alignments 4`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div
        className="item selected"
        onClick={[Function]}>
        <div>
          1
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item "
        onClick={[Function]}>
        <div>
          2
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
    </div>
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly all item alignments 5`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div
        className="item selected"
        onClick={[Function]}>
        <div>
          1
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item "
        onClick={[Function]}>
        <div>
          2
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
    </div>
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly custom ItemContainer 1`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div>
        <div>
          1
        </div>
      </div>
      <div>
        <div>
          2
        </div>
      </div>
    </div>
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly in mobile mode with not known container width 1`] = `
<div
  className="container mobile-device">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div
        className="item selected"
        onClick={[Function]}>
        <div>
          1
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item "
        onClick={[Function]}>
        <div>
          2
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
    </div>
  </div>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly with default props 1`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar" />
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly with many child items 1`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div
        className="item selected"
        onClick={[Function]}>
        <div>
          1
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item "
        onClick={[Function]}>
        <div>
          2
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item "
        onClick={[Function]}>
        <div>
          3
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item "
        onClick={[Function]}>
        <div>
          4
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
    </div>
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly with one child item 1`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div
        className="item selected"
        onClick={[Function]}>
        <div />
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
    </div>
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;

exports[`ScrolledList DOM rendering renders correctly with selected item given arbitrary 1`] = `
<div
  className="container ">
  <div
    className="eyeshot">
    <div
      className="bar">
      <div
        className="item "
        onClick={[Function]}>
        <div>
          1
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
      <div
        className="item selected"
        onClick={[Function]}>
        <div>
          2
        </div>
        <i
          className="KambiWidget-primary-background-color border" />
      </div>
    </div>
  </div>
  <button
    className="left"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(90deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
  <button
    className="right"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "background": "linear-gradient(270deg,  50%, rgba(255, 255, 255, 0))",
      }
    }>
    <svg
      className="icon"
      height="17px"
      version="1.1"
      viewBox="0 860 420 500"
      width="10px"
      xmlns="http://www.w3.org/2000/svg">
      <path
        d="M358.286 640q0-7.429-5.714-13.143l-224.571-224.571 224.571-224.571q5.714-5.714 5.714-13.143t-5.714-13.143l-28.571-28.571q-5.714-5.714-13.143-5.714t-13.143 5.714l-266.286 266.286q-5.714 5.714-5.714 13.143t5.714 13.143l266.286 266.286q5.714 5.714 13.143 5.714t13.143-5.714l28.571-28.571q5.714-5.714 5.714-13.143z"
        id="arrowLeft"
        transform="scale(1, -1) translate(0, -1500)" />
    </svg>
  </button>
</div>
`;
